website/docs/components/form/key-value-inputs/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,148 @@
+
+### Helper text and generic content
+
+The helper text and generic content slot are optional and can be used to provide additional information that applies to all rows.
+
+!!! DO
+
+Use the helper text and generic content slot to reduce the amount of repeated information in each row. This focuses the user’s attention on data submission.
+
+![Helper text used as a means to explain what the toggle in each row does. Generic slot used to include a button with more details.](/assets/components/form/key-value-inputs/key-value-inputs-helper-generic-do.png)
+
+!!!
+
+!!! DONT
+
+Don’t repeat information in each row if it can be explained once at the top of the rows.
+
+![Each row has an input, toggle with helper text enabled and a button in the generic slot. This has repetitive information in each row that over complicates the key value input component's usage.](/assets/components/form/key-value-inputs/key-value-inputs-helper-generic-dont.png)
+
+!!!
+
+## Input rows
+
+Each row often has two inputs, one for a key and one for a value. However, there may be exceptions where a single input may be necessary. 
+
+![](/assets/components/form/key-value-inputs/key-value-inputs-value-only.png)
+
+### Input types
+
+Key and value inputs can use various form elements, although there are some common patterns that represent the majority of HashiCorp use cases.
+
+For the key input, this includes:
+
+- Text Input
+- Select
+
+For the value input, this includes:
+
+- Text Input
+- File Input
+- SuperSelect
+- Masked Input
+
+### Adding rows
+
+Users can add rows to the set of existing rows via the add button.
+
+![A user clicking the add user button where it shows one row of inputs that turns into two. The single row of inputs doesn't have a delete button.](/assets/components/form/key-value-inputs/key-value-inputs-adding-row.png)
+
+#### Hitting a row limit
+
+If there is a row limit and the user reaches it, the add button should be replaced with an inline alert notifying them of the limit.
+
+![After the last input row, there is a neutral alert that says "Limit of four invites has been reached."](/assets/components/form/key-value-inputs/key-value-inputs-row-limit.png)
+
+If the maximum number of rows has been reached and the user deletes a row, the alert should be replaced with the add button.
+
+### Delete behavior
+
+!!! Info
+
+The interactions described in this section will have to be implemented by the consumer.
+
+!!!
+
+A single row is required at all times. If that row's content is empty, the delete button will be hidden.
+
+![A single row with a key and value input with no delete button on the side.](/assets/components/form/key-value-inputs/key-value-inputs-single-row-empty-content.png)
+
+Once content is filled in either of the two inputs, the delete button will become visible. Clicking it will delete that row's data.
```

Comment:
```suggestion
Once content is entered in either of the two inputs, the delete button will become visible. Clicking it will delete the row and its data.
```